netlogo-gui/src/main/app/ChangeManager.scala

```diff
@@ -0,0 +1,138 @@
+package org.nlogo.app
+
+import java.util.List
+import java.util.ArrayList
+import javax.swing.event.ChangeEvent
+/**
+  * Manages a Queue of Changables to perform undo and/or redo operations. Clients can add implementations of the Changeable
+  * class to this class, and it will manage undo/redo as a Queue.
+  *
+  * @author Greg Cope
```

Comment:
Can you provide the URL of the site where you found this? We need to be sure when we pull in code that it's appropriate for us to be using it.